packages/format-library/src/link/inline.js

```diff
@@ -245,6 +247,10 @@ function InlineLinkUI( {
 		);
 	}
 
+	if ( ! hasLink ) {
+		return null;
+	}
```

Comment:
Without this there will be a flash of "Empty link" because the UI can only be closed via a "click" event on the rich text. As this click event happens on `useLayoutEffect` it will always be slightly late to hide the UI.

Reply:
Since this is the `InlineLinkUI` component this makes sense to me.